dspace-spring-rest/src/main/java/org/dspace/app/rest/RestResourceController.java

```diff
@@ -226,4 +264,180 @@ private void checkModelPluralForm(String apiCategory, String model) {
 			throw new RepositoryNotFoundException(apiCategory, model);
 		}
 	}
+	
+	@RequestMapping(method = RequestMethod.GET, value="/search")
+	ResourceSupport listSearchMethods(@PathVariable String apiCategory, @PathVariable String model) {
+		ResourceSupport root = new ResourceSupport();
+		DSpaceRestRepository repository = utils.getResourceRepository(apiCategory, model);
+		boolean searchEnabled = false;
+		for (Method method : repository.getClass().getMethods()) {
```

Comment:
Because the `RestResourceController` is getting quite big, would it be possible to move all "reflection" related logic to a `DSpaceRestReflectionHelper` class (or something similar)? For example this for-loop could be moved to a method `List<String> getSearchMethodNames(DSpaceRestRepository repository)` after the names can be used to construct links.

Reply:
Good suggestion thanks. I'm going to rebase the PR on the current master and refactor the code